core/src/main/java/hudson/model/UserPropertyListener.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * The MIT License
+ *
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package hudson.model;
+
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+import hudson.model.UserProperty;
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nonnull;
+
+/**
+ * Listener interface which all other user property-specific event-listeners make use of.
+ */
+public interface UserPropertyListener extends ExtensionPoint {
+
+    static final Logger LOGGER = Logger.getLogger(UserPropertyListener.class.getName());
+
+    /**
+     * Fired when a new user property has been created.
+     *
+     * @param username the user
+     * @param value property that was newly created.
+     *
+     */
+    default void onCreated(@Nonnull String username, @Nonnull UserProperty value) {
+        LOGGER.log(Level.FINE, MessageFormat.format("new {0} property created for user {1}", value.getClass().toString(), username));
```

Comment:
My suggestion on syntax when using `java.util.logging` is to take advantage of the lambda forms and use normal string concatenation. In this line, that would be:

```suggestion
        LOGGER.fine(() -> "new " + value.getClass().toString() + " property created for user " + username));
```